docs/channel.md

```diff
@@ -424,6 +424,47 @@ Y
 
 See also: [fromList](#fromlist) factory method.
 
+(channel-topic)=
+
+### topic
+
+:::{versionadded} 23.11.0-edge
+:::
+
+:::{note}
+This feature requires the `nextflow.preview.topic` feature flag to be enabled.
+:::
+
+The `topic` method is used to create a "topic" channel, which is a queue channel that can receive items from many sources implicitly based on a "topic" identifier.
+
+Any process can send items to a topic by using the `topic` option on an output:
+
+```groovy
+process foo {
+  output:
+  val('foo'), topic: 'my-topic'
```

Comment:
Let me be provocative here, keen on your views @bentsherman @pditommaso and probably also @ewels .

This currently proposed syntax does not convince me vs DSL2:
```
output:
val('foo'), topic: 'my-topic'
```
It looks to me like a silent partial going back to DSL1 style.

Instead DSL2 specifies channel connections between processes within the `workflow` block, not the `process` block, and hence I think the main legal syntax should be along the lines:
```
output:
val('foo'), emit: 'foo_val'

[..]

workflow {
    foo.foo_val.topic('my-topic')
}
```

And then, eventually (but not necessarily) also allow the following, more DSL1 styled variant (where the `emit` attribute is not needed if no other process requires this as input):
```
output:
val('foo'), emit: 'foo_val', topic: 'my-topic'
```

Reply:
This sort of DSL1-esque inconsistency became apparent to me while looking at Ben's test implementation in rnaseq:  https://github.com/nf-core/rnaseq/pull/1109#issuecomment-1808450986

Why that outlier syntax line?

Reply:
That is an interesting approach, specifying the topic with the process invocation instead of the definition. It would keep the process def modular and be more flexible -- you could send any process output to any topics you want without patching the module, and you could use emits and topics together.

The only downside is that it's more verbose, although still simpler than the current approach used by nf-core

I think I am keen to try it, but curious to see what others think..

Reply:
> It looks to me like a silent partial going back to DSL1 style.

Possibly, but this is the key point because it decouples the topic from the channel compositions. 

Having a convention that all processes in, for example, nf-core, use `versions` to track the tool version, it's enough to add a new module and the receiver will get the module tool version without doing anything else

Reply:
Thanks for the feedbacks. I get your main point about conciseness. Indeed I see how my proposition results in a verbosity that scales with the number of writers - vs your approach, that just adds one module.

1. To bring things together, I am citing @bentsherman 's suggestion above of having `Channel.fromTopic()`. I think this has the potential to ultimately define the new functionality not as a new channel type, but indeed as a new way to define a queue channel.  You **label** outputs by topic, and then you use the new `fromTopic` to create a channel out of all those outputs with that given label. This way I think it essentially clarifies that, as Ben's says, this is a way of implicitly composing multiple `mix`es.

2. As a second consequence of this picture, I would then see the topic specifier in the process definition as a way to apply the *topic* label (as defined above) to the output:
```
output:
val('foo'), topic: 'my-topic'
```
hence, I would still suggest to consider such `topic` as an additional optional attribute to the output, in addition/alternative to the `emit` specifier, rather than in strict replacement for it. This would allow cases where the output has to be consumed also through non-topic channels. 

(as a final note, both the definition of `fromTopic`, and the definition of `topic` as an additional attribute aside with `emit` seem to me more consistent with DSL2 compared to the starting point - from my point of view)

Reply:
I'm fine to stay with "topic" channel definition, but they should be documented as a new channel type because they have different semantics (many writers). 

This should it should be `channel.topic`, not `fromTopic`

Reply:
> Having a convention that all processes in, for example, nf-core, use versions to track the tool version, it's enough to add a new module and the receiver will get the module tool version without doing anything else

I agree with Paolo on this point. I think it is better to use the current approach which is the least verbose and has a narrow use case, i.e. for clear conventions like nf-core tool versions. There are other use cases like collecting logs for multiqc which would require more flexibility to be a channel topic, but at that point I think you might as well just use explicit channel logic.

Reply:
yep, agree on maintaining the current approach.

I would still suggest to consider such `topic` as an additional optional attribute to the output, in **addition/alternative** to the `emit` specifier, rather than in strict replacement for it. This would allow cases where the output has to be consumed also through non-topic channels.

Reply:
Actually `emit` and `topic` already can be used together, I must have misread something. I will update the docs and integration test